src/power10.rs

```diff
@@ -0,0 +1,125 @@
+use Integer;
+
+/// Provides methods to compute functions related to powers of 10.
+pub trait Power10: Integer {
+
+    /// Returns `true` if the number is a power of 10.
+    ///
+    /// # Examples
+    ///
+    /// ~~~
+    /// use num_integer::Power10;
+    /// assert_eq!(100u32.is_power_of_ten(), true);
+    /// assert_eq!(4u32.is_power_of_ten(), false);
+    /// ~~~
+    fn is_power_of_ten(&self) -> bool;
+}
+
+/// Returns `true` if the number is a power of 10.
+#[inline]
+pub fn is_power_of_ten<T: Power10>(x: T) -> bool {
+    x.is_power_of_ten()
+}
+
+// implementation note: the is_power_of_ten algorithm is based on a perfect hash setup with very simple hash functions
```

Comment:
Did you use any particular tool to create these perfect hash functions?

Reply:
I have some code that does that. Nothing clever. It just picks random values and produces a hash until it finds something.